linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/shared.yml

```diff
@@ -1,14 +1,72 @@
-# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv,multi_platform_sle
+# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv
 # reboot = false
 # strategy = configure
 # complexity = low
 # disruption = medium
-- name: "Prevent Log In to Accounts With Empty Password - system-auth"
-  replace:
-    dest: /etc/pam.d/system-auth
-    regexp: 'nullok'
+- name: Check if system relies on authselect
+  ansible.builtin.stat:
+    path: /usr/bin/authselect
+  register: result_authselect_present
+
+- name: Check integrity of authselect current profile
+  ansible.builtin.command:
+    cmd: authselect check
+  register: result_authselect_check_cmd
+  ignore_errors: yes
+  when:
+    - result_authselect_present.stat.exists
+
+- name: Get authselect current profile
+  ansible.builtin.shell:
+    cmd: authselect current -r | awk '{ print $1 }'
+  register: result_authselect_profile
+  when:
+    - result_authselect_check_cmd is failed
+
+- name: Get authselect current features
+  ansible.builtin.shell:
+    cmd: authselect current | tail -n+3 | awk '{ print $2 }'
+  register: result_authselect_features
+  when:
+    - result_authselect_check_cmd is failed
+
+- name: Define the current authselect profile as a local fact
+  ansible.builtin.set_fact:
+    authselect_profile: "{{ result_authselect_profile.stdout }}"
+  when:
+    - result_authselect_check_cmd is failed
+    - result_authselect_profile | default(false) | bool
+
+- name: Restore the authselect profile if manually modified
+  ansible.builtin.command:
+    cmd: authselect select {{ authselect_profile }} --force
+  register: result_pam_authselect_restore_profile
+  when:
+    - result_authselect_check_cmd is failed
+    - result_authselect_profile | default(false) | bool
+
+- name: Restore the authselect features if manually modified
+  ansible.builtin.command:
+    cmd: authselect enable-feature {{ item }}
+  register: result_pam_authselect_restore_features
+  loop: "{{ result_authselect_features.stdout_lines }}"
+  when:
+    - result_pam_authselect_restore_profile
+    - result_authselect_features | default(false) | bool
+
+- name: "Ensure nullok property is not present via authselect tool"
```

Comment:
This task is not converging, it always reports as `changed.

Reply:
Improved the conditional for this task. Fixed.